### ISSUE TITLE
🚀 Feature Request: Update the typescript worker template to include env and context parameters

### ISSUE THREAD

**Issue by timabb031** [cloudflare/wrangler2#1112]
### Describe the solution

I created a worker using `wrangler init myworker` and answered yes to "Would you like to use TypeScript?" and "Would you like to create a Worker at myworker\src\index.ts?".  Upon opening index.ts I noticed that the default fetch function signature is `async fetch(request: Request): Promise<Response>` which lacks the `env` and `context` parameters I've seen in the docs.

I'm pretty sure I'd want to use a binding to KV/DO/R2 more often that not, so my suggestion is to enhance the default template to include the `env` and `context` parameters so users don't have to trawl through the docs to find the parameters and their TS types.

As typescript is being used `env` would need an interface, so it would be cool if a env.ts file could be added to the template with some comments detailing how I could modify the file to reference KV/etc.

```
/**
 * Learn more at https://developers.cloudflare.com/workers/runtime-apis/kv/#referencing-kv-from-durable-objects-and-workers-using-modules-syntax
 */

export interface Env {
  // example binding to KV
  //YOURKVNAMESPACE: KVNamespace;
}
```

I haven't had time to look into what TS type should be used for the `context` parameter, but it would be useful to have that in the default template too.

**Comment by petebacondarwin** [comment]
This sounds like a good improvement. Would you like to provide a PR @timabb031?

I think that, to keep the template simple, we could just add the `Env` interface definition to the top of the `index.ts`, OK?

**Comment by timabb031** [comment]
@petebacondarwin yes that sound great, I'll try get a PR opened for this in the next week or so.
